### PR TITLE
Implement CRUISE MODE for Fixed Wing

### PR DESCRIPTION
**Pull request by giacomo892**
Implementing `CRUISE MODE` (2D and 3D) for Fixed Wing aircrafts. 

2D Mode: The aircraft will maintain the current heading unless pilot changes it via yaw input or roll the aircraft.

3D Mode: The aircraft will maintain the current altitude (change via pitch command) and the current heading unless pilot changes it via yaw input or roll the aircraft.

It can be enabled together with launch mode so the aircraft will enter in `CRUISE MODE` right after the launch is over.

Thanks to @shellixyz for the contribution to the code and the massive amount of time spent in testing and fix issues with the implementation.

As today the code it is being flown and works but indeed code need a review.

### CODE REVIEW

File: src/main/fc/fc_msp_box.c [hunk 1]
```diff
@@ -188,6 +189,7 @@ void initActiveBoxIds(void)
         activeBoxIds[activeBoxIdCount++] = BOXNAVWP;
         activeBoxIds[activeBoxIdCount++] = BOXHOMERESET;
         activeBoxIds[activeBoxIdCount++] = BOXGCSNAV;
+        if(STATE(FIXED_WING)) {activeBoxIds[activeBoxIdCount++] = BOXNAVCRUISE;} //Temp fix before CRUISE will be active for MC
```
Comment (fiam) [inline src/main/fc/fc_msp_box.c]: Please, move the if body into its own line.

File: src/main/navigation/navigation_fixedwing.c [hunk 2]
```diff
@@ -340,7 +344,7 @@ void applyFixedWingPositionController(timeUs_t currentTimeUs)
                 // Calculate virtual position target at a distance of forwardVelocity * HZ2S(POSITION_TARGET_UPDATE_RATE_HZ)
                 // Account for pilot's roll input (move position target left/right at max of max_manual_speed)
                 // POSITION_TARGET_UPDATE_RATE_HZ should be chosen keeping in mind that position target shouldn't be reached until next pos update occurs
-                // FIXME: verify the above
+                // FIXME: verify the above-
```
Comment (fiam) [inline src/main/navigation/navigation_fixedwing.c]: Can you remove the extra `-`? Seems it's the only change in this line.

**Comment by digitalentity** [comment]
@giacomo892 @shellixyz rebase is required

**Comment by shellixyz** [comment]
Rebased and squashed. I tested this extensively it is ready to merge IMO.

**Review by digitalentity — APPROVED** [review]

**Comment by digitalentity** [comment]
Squashed again and merged. Thanks!

**Pull request merged**